### PR TITLE
[core] unit tests and tweaks for Category

### PR DESCRIPTION
**Pull request by jamesonwilliams**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/main/java/com/amplifyframework/core/category/Category.java [hunk 1]
```diff
@@ -61,44 +62,38 @@ public Category() {
...
-            String pluginKey = plugin.getPluginKey();
-            JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
+            for (P plugin : getPlugins()) {
+                String pluginKey = plugin.getPluginKey();
+                JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
+                plugin.configure(pluginConfig, context);
+            }
 
-            plugin.configure(pluginConfig, context);
+            configurationState.set(ConfigurationState.CONFIGURED);
```
Comment (raphkim) [inline core/src/main/java/com/amplifyframework/core/category/Category.java]: shouldn't this be set outside the `for` loop?

Comment (jamesonwilliams) [reply core/src/main/java/com/amplifyframework/core/category/Category.java]: YES! Thanks, will fix.

Comment (raphkim) [reply core/src/main/java/com/amplifyframework/core/category/Category.java]: oh wow that was really confusing LOL

**Comment by jamesonwilliams** [comment]
Moving this out of my fork for now, due to CircleCI issues.

**Pull request closed without merging**